### DISCUSSION TITLE
build error : obsolete classes

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/runtime#63461] [Q&A]
Hello
I get below errors when built runtime libraries. 
"build cmd clr+libs -rc Release"
How we can fix ? 
Thanks

C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1460,67): error SYSLIB0017: 'StrongNameKeyPair' is obsolete: 'Strong name signing is not supported and throws PlatformNotSupportedException.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1523,67): error SYSLIB0015: 'DisablePrivateReflectionAttribute' is obsolete: 'DisablePrivateReflectionAttribute has no effect in .NET 6.0+.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1574,67): error SYSLIB0025: 'SuppressIldasmAttribute' is obsolete: 'SuppressIldasmAttribute has no effect in .NET 6.0+.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1585,67): error SYSLIB0004: 'Cer' is obsolete: 'The Constrained Execution Region (CER) feature is not supported.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1586,67): error SYSLIB0004: 'Consistency' is obsolete: 'The Constrained Execution Region (CER) feature is not supported.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1588,67): error SYSLIB0004: 'PrePrepareMethodAttribute' is obsolete: 'The Constrained Execution Region (CER) feature is not supported.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1589,67): error SYSLIB0004: 'ReliabilityContractAttribute' is obsolete: 'The Constrained Execution Region (CER) feature is not supported.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1592,67): error SYSLIB0032: 'HandleProcessCorruptedStateExceptionsAttribute' is obsolete: 'Recovery from corrupted process state exceptions is not supported; HandleProcessCorruptedStateExceptionsAttribute is ignored.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1596,67): error SYSLIB0003: 'AllowReversePInvokeCallsAttribute' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1839,67): error SYSLIB0021: 'AesCryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1841,67): error SYSLIB0021: 'AesManaged' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1862,67): error SYSLIB0021: 'DESCryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1892,67): error SYSLIB0021: 'MD5CryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1904,67): error SYSLIB0021: 'RC2CryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1906,67): error SYSLIB0022: 'Rijndael' is obsolete: 'The Rijndael and RijndaelManaged types are obsolete. Use Aes instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1907,67): error SYSLIB0022: 'RijndaelManaged' is obsolete: 'The Rijndael and RijndaelManaged types are obsolete. Use Aes instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1908,67): error SYSLIB0023: 'RNGCryptoServiceProvider' is obsolete: 'RNGCryptoServiceProvider is obsolete. To generate a random number, use one of the RandomNumberGenerator static methods instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1923,67): error SYSLIB0021: 'SHA1CryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1924,67): error SYSLIB0021: 'SHA1Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1926,67): error SYSLIB0021: 'SHA256CryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1927,67): error SYSLIB0021: 'SHA256Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1929,67): error SYSLIB0021: 'SHA384CryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1930,67): error SYSLIB0021: 'SHA384Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1932,67): error SYSLIB0021: 'SHA512CryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1933,67): error SYSLIB0021: 'SHA512Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1938,67): error SYSLIB0021: 'TripleDESCryptoServiceProvider' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1981,67): error SYSLIB0003: 'IPermission' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1984,67): error SYSLIB0003: 'CodeAccessSecurityAttribute' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1985,67): error SYSLIB0003: 'SecurityAction' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1986,67): error SYSLIB0003: 'SecurityAttribute' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1987,67): error SYSLIB0003: 'SecurityPermissionAttribute' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]
C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\netstandard\net7.0-Debug\netstandard.Forwards.cs(1988,67): error SYSLIB0003: 'SecurityPermissionFlag' is obsolete: 'Code Access Security is not supported or honored by the runtime.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\shims\generated\netstandard.csproj]

**Comment by huoyaoyuan** [comment]
I can't reproduce this. You can delete the `artifacts` folder and see if it reproduces with a clean build. Typically it's caused by inconsistency in intermediate file.

**Reply by ghost** [reply]
Hi, thanks for your reply.
At first build, I get "Derived cryptographic types are obsolete" errors. Also I reported at here : https://github.com/dotnet/runtime/discussions/63449 And then I comment out these obsolete lines in source and started build again. Then I get the erros which explained at this topic.

I delete "artifacts" folder and start build again but the same errros occured. 

This is my first time building runtime project.

**Reply by huoyaoyuan** [reply]
I did clean build several times these days, and no problem occurs. You should not need to modify any code.
Note: the `[Obsolete]` exist in both `src` and `ref` folders. The generated code shouldn't be impacted by SYSLIB warnings.

Do you close any IDE instance when invoking the build script?

**Reply by ghost** [reply]
Hi, at first I cloned 'master' branch via visual studio. 
Then I run this command "build.cmd -subset clr" to build CLR project. It builded successfully. 
Then I wanted to build all libraries so I run this command "build.cmd clr+libs -rc Release" as instructions said.
But I get below errors : 

C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(10,67): error SYSLIB0021: 'AesManaged' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]
C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(55,67): error SYSLIB0022: 'Rijndael' is obsolete: 'The Rijndael and RijndaelManaged types are obsolete. Use Aes instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]
C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(56,67): error SYSLIB0022: 'RijndaelManaged' is obsolete: 'The Rijndael and RijndaelManaged types are obsolete. Use Aes instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]
C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(58,67): error SYSLIB0021: 'SHA1Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]
C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(60,67): error SYSLIB0021: 'SHA256Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]
C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(62,67): error SYSLIB0021: 'SHA384Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]
C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.Forwards.cs(64,67): error SYSLIB0021: 'SHA512Managed' is obsolete: 'Derived cryptographic types are obsolete. Use the Create method on the base type instead.' [C:\Users\mrkiyak\source\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj]

**Reply by huoyaoyuan** [reply]
@dotnet/area-infrastructure-libraries Could you provide any help?

**Reply by safern** [reply]
@lepptin could you please add `/bl` to your command? Then once the build fails, please share with me the msbuild.binlog file that was created on the root of your repository clone. That will give me more information to see why you might be hitting these issues.

**Reply by ghost** [reply]
[msbuild.zip](https://github.com/dotnet/runtime/files/7849503/msbuild.zip)
@safern Hi, I've just added the log file. Thanks.

**Reply by safern** [reply]
Thanks, @lepptin. I've looked at it. While I don't have a solution yet. Could you try and run the following on a command line under the path: `<your-repo>\src\libraries\System.Security.Cryptography.Algorithms\ref`.

```cmd
"C:\Program Files\dotnet\dotnet.exe" exec "C:\Program Files\dotnet\sdk\6.0.100\Roslyn\bincore\csc.dll" /noconfig /unsafe- /checked- /nowarn:1701,1702,1705,1591,CS8969,CS0169,CS0649,CS8618,CS8597,CS8625,CS8617,"SYSLIB0021","SYSLIB0022",NU5105,1701,1702 /fullpaths /nostdlib+ /errorreport:prompt /warn:9999 /define:TRACE;DEBUG;NET;NET7_0;NETCOREAPP;NET5_0_OR_GREATER;NET6_0_OR_GREATER;NETCOREAPP1_0_OR_GREATER;NETCOREAPP1_1_OR_GREATER;NETCOREAPP2_0_OR_GREATER;NETCOREAPP2_1_OR_GREATER;NETCOREAPP2_2_OR_GREATER;NETCOREAPP3_0_OR_GREATER;NETCOREAPP3_1_OR_GREATER /highentropyva+ /nullable:enable /reference:C:\Users\mrkiyak\source\repos\runtime\artifacts\bin\System.Runtime\ref\net7.0-Debug\System.Runtime.dll /reference:C:\Users\mrkiyak\source\repos\runtime\artifacts\bin\System.Security.Cryptography\ref\net7.0-Debug\System.Security.Cryptography.dll /debug+ /debug:portable /delaysign- /filealign:512 /keyfile:C:\Users\mrkiyak\.nuget\packages\microsoft.dotnet.arcade.sdk\7.0.0-beta.22053.2\tools\snk/MSFT.snk /optimize- /out:C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\System.Security.Cryptography.Algorithms.dll /target:library /warnaserror+ /utf8output /deterministic+ /publicsign+ /checksumalgorithm:SHA256 /langversion:preview /skipanalyzers+ /features:strict /features:nullablePublicOnly /analyzerconfig:C:\Users\mrkiyak\source\repos\runtime\.editorconfig /analyzerconfig:C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\System.Security.Cryptography.Algorithms.GeneratedMSBuildEditorConfig.editorconfig /analyzer:"C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\..\analyzers\ILLink.CodeFixProvider.dll" /analyzer:"C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\..\analyzers\ILLink.RoslynAnalyzer.dll" System.Security.Cryptography.Algorithms.Forwards.cs "C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\.NETCoreApp,Version=v7.0.AssemblyAttributes.cs" "C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\_AssemblyInfo.cs" "C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\System.Security.Cryptography.Algorithms.AssemblyInfo.cs" /warnaserror+:NU1605
```

**Reply by ghost** [reply]
Hi @safern Could you please post the command line again. I couldn't get

**Reply by safern** [reply]
Sure, it is here:

> "C:\Program Files\dotnet\dotnet.exe" exec "C:\Program Files\dotnet\sdk\6.0.100\Roslyn\bincore\csc.dll" /noconfig /unsafe- /checked- /nowarn:1701,1702,1705,1591,CS8969,CS0169,CS0649,CS8618,CS8597,CS8625,CS8617,"SYSLIB0021","SYSLIB0022",NU5105,1701,1702 /fullpaths /nostdlib+ /errorreport:prompt /warn:9999 /define:TRACE;DEBUG;NET;NET7_0;NETCOREAPP;NET5_0_OR_GREATER;NET6_0_OR_GREATER;NETCOREAPP1_0_OR_GREATER;NETCOREAPP1_1_OR_GREATER;NETCOREAPP2_0_OR_GREATER;NETCOREAPP2_1_OR_GREATER;NETCOREAPP2_2_OR_GREATER;NETCOREAPP3_0_OR_GREATER;NETCOREAPP3_1_OR_GREATER /highentropyva+ /nullable:enable /reference:C:\Users\mrkiyak\source\repos\runtime\artifacts\bin\System.Runtime\ref\net7.0-Debug\System.Runtime.dll /reference:C:\Users\mrkiyak\source\repos\runtime\artifacts\bin\System.Security.Cryptography\ref\net7.0-Debug\System.Security.Cryptography.dll /debug+ /debug:portable /delaysign- /filealign:512 /keyfile:C:\Users\mrkiyak\.nuget\packages\microsoft.dotnet.arcade.sdk\7.0.0-beta.22053.2\tools\snk/MSFT.snk /optimize- /out:C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\System.Security.Cryptography.Algorithms.dll /target:library /warnaserror+ /utf8output /deterministic+ /publicsign+ /checksumalgorithm:SHA256 /langversion:preview /skipanalyzers+ /features:strict /features:nullablePublicOnly /analyzerconfig:C:\Users\mrkiyak\source\repos\runtime\.editorconfig /analyzerconfig:C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\System.Security.Cryptography.Algorithms.GeneratedMSBuildEditorConfig.editorconfig /analyzer:"C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\..\analyzers\ILLink.CodeFixProvider.dll" /analyzer:"C:\Program Files\dotnet\sdk\6.0.100\Sdks\Microsoft.NET.Sdk\targets\..\analyzers\ILLink.RoslynAnalyzer.dll" System.Security.Cryptography.Algorithms.Forwards.cs "C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\.NETCoreApp,Version=v7.0.AssemblyAttributes.cs" "C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\_AssemblyInfo.cs" "C:\Users\mrkiyak\source\repos\runtime\artifacts\obj\System.Security.Cryptography.Algorithms\ref\net7.0-Debug\System.Security.Cryptography.Algorithms.AssemblyInfo.cs" /warnaserror+:NU1605

**Reply by ghost** [reply]
Hi @safern unfortunately I couldn't. get this error on CLI "error CS2001: Source file 'C:\Users\mrkiyak\source\repos\runtime.editorconfig' could not be found"

**Reply by safern** [reply]
@lepptin it seems like your clone is on a weird state. I would recommend doing the following. 

```
git clean -fdx
git fetch <dotnet-upstream>/main
git checkout dotnet/main
```

and then try to build again using the build.cmd script.
